cf_units/tests/unit/unit/test_as_unit.py

```diff
@@ -0,0 +1,87 @@
+# (C) British Crown Copyright 2014, Met Office
+#
+# This file is part of cf_units.
+#
+# cf_units is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cf_units is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with cf_units.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for the `cf_units.as_unit` function."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+import copy
+import unittest
+
+import mock
+
+import cf_units
+from cf_units import as_unit, Unit
+
+
+class StubUnit(object):
```

Comment:
Thanks @cpelley - this is highlighting my ignorance, as well as the problem with "see x" type descriptions.

Essentially, what you are doing in this PR is enabling duck-typing of units.

Perhaps you could add a comment (which concretely mentions iris.unit.Unit) to this class, so that we can see the motivation for its existence.